### PR TITLE
feat: Release `kedro-datasets` version `1.3.0`

### PR DESCRIPTION
**Pull request by jmholzer**
## Description
Release datasets version 1.3.0

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by noklam — APPROVED** [review]
Lgtm

File: kedro-datasets/RELEASE.md [hunk 1]
```diff
@@ -1,12 +1,21 @@
 # Upcoming Release:
 
-## Major features and improvements:
+## Major features and improvements
+
+## Bug fixes and other changes
+
+## Community contributions
+
+# Release 1.3.0:
```
Comment (astrojuanlu) [inline kedro-datasets/RELEASE.md:9]:
```suggestion
# Release 1.3.0
```

?

File: kedro-datasets/RELEASE.md [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
 * Added pandas 2.0 support.
 * Added SQLAlchemy 2.0 support (and dropped support for versions below 1.4).
 * Added a save method to the APIDataSet
 * Reduced constructor arguments for `APIDataSet` by replacing most arguments with a single constructor argument `load_args`. This makes it more consistent with other Kedro DataSets and the underlying `requests` API, and automatically enables the full configuration domain: stream, certificates, proxies, and more.
 * Relaxed Kedro version pin to `>=0.16`
 * Added `metadata` attribute to all existing datasets. This is ignored by Kedro, but may be consumed by users or external plugins.
+* Added `ManagedTableDataSet` for managed delta tables on Databricks.
 
 ## Bug fixes and other changes
 * Relaxed `delta-spark` upper bound to allow compatibility with Spark 3.1.x and 3.2.x.
```
Comment (astrojuanlu) [inline kedro-datasets/RELEASE.md:21]: Never added this to the release notes but gh-171:

```suggestion
* Relaxed `delta-spark` upper bound to allow compatibility with Spark 3.1.x and 3.2.x.
* Upgraded required `polars` version to 0.17.
```

**Review by SajidAlamQB — APPROVED** [review]

File: kedro-datasets/RELEASE.md (hunk 1 above)
Comment (jmholzer) [reply kedro-datasets/RELEASE.md:9]: To be consistent with the rest of the file, it looks like `Release x.x.x` should have a colon, no other titles / subtitles should. Now that you mention it, `Release 1.2.0` is missing a colon too :).

**Pull request merged**